src/hotspot/share/gc/g1/g1ParScanThreadState.cpp

```diff
@@ -712,6 +712,5 @@ G1ParScanThreadStateSet::~G1ParScanThreadStateSet() {
   assert(_flushed, "thread local state from the per thread states should have been flushed");
   FREE_C_HEAP_ARRAY(G1ParScanThreadState*, _states);
   FREE_C_HEAP_ARRAY(size_t, _surviving_young_words_total);
-  _preserved_marks_set.assert_empty();
```

Comment:
The assert checks that the preserved marks have been restored, so it does not seem unnecessary at first glance.